test/express.test.js

```diff
@@ -37,9 +36,85 @@ describe('Server tests', function () {
+    const response =  await fetch(url);
+    const data = await response.json();
+    
     expect(response.status).to.equal(200);
     expect(response.headers.get('content-type')).to.include('application/json');
     expect(data).to.deep.equal(expectedData);
   });
+
+  it('returns correct data when provided no file', async function() {
+    testServer = new Server(['./servers/express.js','http:127.0.0.1:8080']);
+    await testServer.readyPromise;
+    
+    const response =  await fetch('http://127.0.0.1:8080/api');
+    const data = await response.json();
+    
+    const expectedData = {
+      range: {},
+      timeseries: {
+        eventloop: [],
+        memory: [],
+        cpu: []
+      }
+    };
+
+    expect(response.status).to.equal(200);
+    expect(response.headers.get('content-type')).to.include('application/json');
+    expect(data).to.deep.equal(expectedData);
+  });
+
+  it('only allows text files to be uploaded', async function() {
+    testServer = new Server(['./servers/express.js','http:127.0.0.1:8080']);
+    await testServer.readyPromise;
+    
+    // upload a couple files
+    const form = new FormData();
+    form.append('file', fs.createReadStream(path.join(__dirname, 'sample-data.log')));
+    form.append('file', fs.createReadStream(path.join(__dirname, 'servers', 'express.js')));
+
+    const response =  await fetch('http://127.0.0.1:8080/api/upload', {method: 'POST', body: form});
+    const data = await response.json();
+    
+    // check if everything's right
+    expect(response.status).to.equal(200);
+    expect(data.files.length).to.equal(1);
+    expect(data.files[0].mimetype).to.equal('text/plain');
+
+    // delete any files created during the test
+    fs.rm(path.join(__dirname, '..', 'uploads', data.files[0].filename), () => {});
+  });
+
+  it('starts watching a new logfile when needed', async function() {
+    testServer = new Server(['./servers/express.js','http:127.0.0.1:8080']);
+    await testServer.readyPromise;
+
+    // upload a file
+    const form = new FormData();
+    form.append('file', fs.createReadStream(path.join(__dirname, 'sample-data.log')));
+
+    let response = await fetch('http://127.0.0.1:8080/api/upload', {method: 'POST', body: form});
+    let data = await response.json();
+    
+    // set the new logfile
+    const newLogfile = data.files[0].filename;
+    response = await fetch('http://127.0.0.1:8080/api/watchfile', {
+      method: 'POST',
+      headers: {'Content-Type': 'application/json'},
+      body: JSON.stringify({filename: newLogfile}),
+    });
+
+    // check if everything's good
+    response = await fetch('http://127.0.0.1:8080/api/curr-logfile');
+    data = await response.json();
+
+    expect(response.status).to.equal(200);
+    expect(data.currentLogfile).to.equal(newLogfile);
+
+    // delete any files created during the test
+    fs.rm(path.join(__dirname, '..', 'uploads', newLogfile), () => {});
```

Comment:
any time this kind of thing is repeated it starts to make sense to either parameterize it or to create a function so the code is not duplicated multiple times.

i'd probably do something like
```
function makeUploadedFilename(filename) {
  return path.join(__dirname, '..', 'uploads', filename);
}
```

also, it's not a good idea to ignore errors that shouldn't happen - you don't know whether you're trying to delete a file that isn't there, that you don't have permissions to delete, or there is a serious error. it's better the test fail if there should be a file present but there is an error deleting it.